[CS2113T-F11-4] SCRUMptious

gmit22 opened the pull request:
SCRUMptious is a command line software for project management designed for project managers following the scrum methodology to develop it.

theopin commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,308 @@
+### Model Component
+{UML}
+### Storage Component
+![Figure X: Simplified class diagram for Storage Component, Model and json.simple](./image/developerguide/storagecomponent.png "Storage Component UML")  
+  
```
![image](https://user-images.githubusercontent.com/52351742/97386218-54ed3380-190e-11eb-84e8-1ed024ea511d.png)

Perhaps you could add association labels to make the diagram more readable for the user.

brandontoh commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,308 @@
+1. [Introduction](#introduction)
+    1. [Background](#background)
+    1. [Purpose](#purpose)
+    1. [Scope](#scope)
+1. [Getting Started](#getting-started)
+    1. [Prerequisites](#prerequisites)
+    1. [Setting Up](#setting-up)
+    1. [Running the Program](#running-the-program)
+1. [Design](#design)
+    1. [Architecture](#architecture)
+    1. [UI Component](#ui-component)
+    1. [Logic Component](#logic-component)
+    1. [Model Component](#model-component)
+    1. [Storage Component](#storage-component)
+1. [Implementation](#implementation)
+    1. [Project](#project)
+        1. [Create Project](#create-project)
+        1. [Select Project](#select-project)
+    1. [Task](#task)
+        1. [Add Task](#add-task)
+        1. [View Task](#view-task)
+        1. [Delete Task](#delete-task)
+        1. [Change Task Priority](#change-task-priority)
+        1. [Mark Task as Complete](#mark-task-as-complete)
+    1. [Sprint](#sprint)
+        1. [Create Sprint](#create-sprint)
+        1. [View Sprint](#view-sprint)
+        1. [Add Task to Sprint](#add-task-to-sprint)
+        1. [Remove Task from Sprint](#remove-task-from-sprint)
+        1. [Allocate Sprint Tasks to Members](#allocate-sprint-tasks-to-members)        
+    1. [Storage](#storage)
+        1. [Location](#location)
+        1. [Loading Data](#loading-data)
+            1. [Converting and Mapping of JSON to Objects](#converting-and-mapping-of-json-to-objects)
+        1. [Saving Data](#saving-data)
+            1. [When the Program Exits](#when-the-program-exits)
+            1. [Changes Made to the Data](#changes-made-to-the-data)
+            1. [Serialising Objects to JSON](#serialising-objects-to-json)
+1. [Appendix: Requirements]()
+1. [Others](#target-user-profile)
```
I feel that it might be better to use numbering like 1.1, 1.2 so that the user would be able to see the hierarchy better

brandontoh commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,308 @@
+### Model Component
+{UML}
+### Storage Component
+![Figure X: Simplified class diagram for Storage Component, Model and json.simple](./image/developerguide/storagecomponent.png "Storage Component UML")  
```
The diagram looks really nice, but i think you could consider adding some multiplicities in as well

theopin commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,308 @@
+### UI Component
+![Figure X: Simplified class diagram for UI Component](./image/developerguide/UI.png "User Interface")  
+### Logic Component
+![Figure X: Simplified class diagram for Logic Component](./image/developerguide/parserandcommand.png)
```
![image](https://user-images.githubusercontent.com/52351742/97386853-beba0d00-190f-11eb-8bad-1ef637bd9e6a.png)

Should a {abstract} be appended to the end of the method declaration for execute() since it seems to be an abstract method

theopin replied:
> ![image](https://user-images.githubusercontent.com/52351742/97386853-beba0d00-190f-11eb-8bad-1ef637bd9e6a.png)
> 
> Should a {abstract} be appended to the end of the method declaration for execute() since it seems to be an abstract method

brandontoh commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,308 @@
+This section seeks to explain the high-level design of the application. Given below is a quick overview of each component and the explanation of the design architecture in greater detail.
+NotUS is the main class of the application, and handles the initializing and execution of the appropriate classes.
+### Architecture
+![Figure X: Architecture Diagram](./image/developerguide/architecturediagram.png "Architecture Diagram UML")  
```
I like how simple and easy your architecture diagram looks, but I am a bit lost when the user points to UI and there's no arrow pointing out of the UI, it feel like the flow wasn't complete. I thought that it could be better if the UI points to the user, it would make the flow continuous as well.

gmit22 replied:
To clarify, there should be two arrows one from the user to the UI and one from UI to the user?

brandontoh commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,308 @@
+Each of the modules listed above are a collection of constituent classes, with each handling specialized tasks in-line with the SLAP principle.
+
+### UI Component
+![Figure X: Simplified class diagram for UI Component](./image/developerguide/UI.png "User Interface")  
```
The UI diagram is easy to understand but there's no description to the diagram yet. However, I feel that you that you can also consider adding a little bit of dependencies, inheritance to the diagram. Or even methods and attributes, that might help you when you are explaining how the UI manager works.

AndreWongZH replied:
Maybe you can add a explanation to clarify what this UI diagram means. From what i can see, The Ui manager class have arrows pointing to its methods. I believe you can recheck the syntax of the class diagram.

theopin commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,308 @@
+When you start the program, the program will load the data file from its respective location and deserialise it into its respective objects. Data will be saved when the program exits or whenever the user makes changes to the program.  
+
+#### Loading Data
+![Figure X: Loading Data](image/developerguide/storage_load.png "Loading Data")  
```
![image](https://user-images.githubusercontent.com/52351742/97387666-6be15500-1911-11eb-9f85-e29ec2427402.png)
Perhaps you could add a dotted arrow line from the second activation bar to the user lifeline to indicate return of control to the user

theopin commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,308 @@
+*`Priority` is an **enum** and is the only model which does not follow this strictly. It is mapped by type casting the property as `String` first, then calling the `Priority.valueOf()` method to convert it into its respective **enum**.  
+
+#### Saving Data
+![Figure X: Saving Data](image/developerguide/storage_save.png "Saving Data")  
```
![image](https://user-images.githubusercontent.com/52351742/97387910-ef9b4180-1911-11eb-9866-9d6346276a76.png)
Perhaps you could add dotted arrow line from the second activation bar to the user lifeline to indicate return of control to the place that called tis seqemce

brandontoh reviewed (commented):
Overall, the diagrams are really nice with appropriate level of details, making it easy to understand.

AndreWongZH commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,308 @@
+### UI Component
+![Figure X: Simplified class diagram for UI Component](./image/developerguide/UI.png "User Interface")  
+### Logic Component
+![Figure X: Simplified class diagram for Logic Component](./image/developerguide/parserandcommand.png)
```
Maybe you can label the diagram. For example Figure 3. Class diagram of logic component.
Also, from looking at your ParserManager, it seems like the ParserManager do not hold any attributes for the other parser class. They are only called in the parser method of the ParserManager. The arrows to the parser might be dotted to indicate a dependency.

AndreWongZH commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,308 @@
+An existing task is deleted when the user wishes to remove a completed or unnecessary task. The task
+ID is provided.
+Command executed by user `task /del -id`
+is passed, the following operations are implemented:
+    * UI receives user input and passes it to Parser class.
+    * Parser checks if the id entered is valid, and executes a corresponding DeleteTaskCommand 
+    object.
+    * The corresponding task is deleted from the program.
```
The bullet point here is not rendered correctly. Perhaps add a space?

AndreWongZH commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,308 @@
+      3. Call `fromJson()` of the newly created object, passing the property as the parameter (e.g. `Sprint.fromJson()`).
+      4. New object's `fromJson()` will **repeat the same process again under Step 4** for its own properties.  
+      
+*`Priority` is an **enum** and is the only model which does not follow this strictly. It is mapped by type casting the property as `String` first, then calling the `Priority.valueOf()` method to convert it into its respective **enum**.  
```
Perhaps you can link the definition of Priority under Glossary instead for better clarity

AndreWongZH commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,308 @@
+To make the data persistent and portable, JSON has been chosen as the format for data to be saved to a persistent storage such as storage drives, thumb drives and any other storage medium which stores the program. JSON is also **human-readable** which allows users to directly modify the data file easily. This can be useful in certain scenarios such as fixing the data file in the event of data cor ...
+
+#### Location
+![Figure X: Running the Jar](image/developerguide/savingdata1.png "Running the Jar")  
+_Figure X: Running the Jar_  
+![Figure X: Running in IDE](image/developerguide/savingdata2.png "Running in IDE")  
+_Figure X: Running in IDE_  
```
Maybe these figures can be side by side to show the difference.

AndreWongZH reviewed (commented):
Overall the explanation seems great. Great job!

hughjazzman commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+
+## Design
+This section seeks to explain the high-level design of the application. Given below is a quick overview of each component and the explanation of the design architecture in greater detail.
+NotUS is the main class of the application, and handles the initializing and execution of the appropriate classes.
```
Is NotUS perhaps possibly a typo? It's not defined previously.

hughjazzman commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+A quick overview of the components is as follows:
+
+`Main` is single-class component of `SCRUMptious`. It is responsible for:
+    1. At app launch: Initializes the components in the correct sequence(Storage, UI, Parser), and links them together where appropriate.
```
Perhaps put this on the next line? It's not rendering properly:
![image](https://user-images.githubusercontent.com/53231968/97394973-e2d11a80-191e-11eb-9bdd-4f942bf9dc00.png)

hughjazzman commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+    1. At app launch: Initializes the components in the correct sequence(Storage, UI, Parser), and links them together where appropriate.
+    1. At program exit: Invokes Storage component to save all unsaved data.
+
+The other packages are described below: 
```
Maybe consider either putting all of these in the diagram, or to leave this section to only explain what's in the diagram? There's possibly a disconnect between the text and the diagram.

dojh111 commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
-## Design & implementation
+## Table of Contents
+1. [Introduction](#introduction)
+    1. [Background](#background)
+    1. [Purpose](#purpose)
+    1. [Scope](#scope)
```
Similar to what was mentioned below, it feel it would be better to number the sub sections in the 1.1, 1.2 format as well.

hughjazzman commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+1. `Parser` : Creates a suitable parser, based on the command to make sense of the user input. Respective parser then
+ make use the information and call respective commands. 
+ 
+Each of the modules listed above are a collection of constituent classes, with each handling specialized tasks in-line with the SLAP principle.
```
Consider explaining the acronym SLAP as it's not defined anywhere?

dojh111 commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+NotUS is the main class of the application, and handles the initializing and execution of the appropriate classes.
+### Architecture
+![Figure X: Architecture Diagram](./image/developerguide/architecturediagram.png "Architecture Diagram UML")  
+The **Architecture Diagram** shown above describes the high level association operations of the application. 
```
I think it will be better if the diagrams are labelled with a number instead, for better readability as the reader now knows which diagram to refer to.

hughjazzman commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+Each of the modules listed above are a collection of constituent classes, with each handling specialized tasks in-line with the SLAP principle.
+
+### UI Component
+![Figure X: Simplified class diagram for UI Component](./image/developerguide/UI.png "User Interface")  
```
Perhaps there could be an explanation of this diagram? It is unclear what type of diagram this is.

dojh111 commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+- `SprintParser`
+- `HelpParser`
+
+The `ProjectParser` validates the parameters of the command. If the command is valid, it returns the respective `ProjectCommandXYZ` to the `ParserManager`.
+If the command is invalid, the `ProjectParser` returns an appropriate warning message to the user.  
+
+The `MemberParser` validates the parameters of the command. If the command is valid, it returns the respective `MemberCommandXYZ` to the `ParserManager`.
+If the command is invalid, the `TaskParser` returns an appropriate warning message to the user.  
+
+The `TaskParser` validates the parameters of the command. If the command is valid, it returns the respective `TaskCommandXYZ` to the `ParserManager`.
+If the command is invalid, the `TaskParser` returns an appropriate warning message to the user.  
+
+The `SprintParser` validates the parameters of the command. If the command is valid, it returns the respective `SprintCommandXYZ` to the `ParserManager`.
+If the command is invalid, the `SprintParser` returns an appropriate warning message to the user.  
+
+The `HelpParser` validates the parameters of the command. If the command is valid, it returns the respective `HelpParserCommandXYZ` to the `ParserManager`.
+If the command is invalid, the `HelpParser` returns an appropriate warning message to the user.
```
I like the way this is styled. However I think it would be better if the different "xxxParser" exceptions parsers were split into different subheadings.

hughjazzman commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+### UI Component
+![Figure X: Simplified class diagram for UI Component](./image/developerguide/UI.png "User Interface")  
+### Logic Component
+![Figure X: Simplified class diagram for Logic Component](./image/developerguide/parserandcommand.png)
```
Nice diagram overall. Maybe there could be more distinction between an arrow and a border? Since they look the same it might be slightly harder to decipher which is which. Consider also adding a description to the diagram in the text.

dojh111 commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+1. `ProjectMembers` : Stores all team-members associated to a project. 
+1. `Parser` : Creates a suitable parser, based on the command to make sense of the user input. Respective parser then
+ make use the information and call respective commands. 
+ 
```
Maybe a simple sequence diagram to show how all the different components of you app interact with each other and the user would help make things clearer on what happens within the program when a generic command is run.

hughjazzman commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+
+When a user types a command, `SCRUMptious`calls the `ParserManager`. The `ParserManager` then parses commands from the user. Subsequently, 
+the `ParserManager` passes the commands on to the respective exceptions parsers which inherit from the `ExceptionsParser` interface. 
+The exceptions parsers consist of:  
```
Perhaps these could use `*` for the bullet points to follow the standard? https://se-education.org/guides/conventions/markdown.html

Same goes for the rest of the document as well.

hughjazzman commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+* Includes a SprintList to manage every Sprints created by the user
+* Sprint can contain Tasks and Members allocated to those Tasks
+### Storage Component
+![Figure X: Simplified class diagram for Storage Component, Model and json.simple](./image/developerguide/storagecomponent.png "Storage Component UML")  
```
Not a big issue, but perhaps `0..*` could be replaced with `*`?
![image](https://user-images.githubusercontent.com/53231968/97395576-355f0680-1920-11eb-9035-47a9377a7d05.png)

dojh111 replied:
Very clean diagram. One recommendation I would suggest is to add some association labels on some of the arrows to show how the 2 classes are associated with each other. Also, should the implementation of the interface from the 2 Storage classes to the Json.simple interface be a dotted line?

hughjazzman commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+
+#### Create Sprint
+
+![Figure X: Sequence diagram of CreateSprintCommand](./image/developerguide/createSprint.png "Create Sprint Sequence Diagram")  
```
Perhaps there could be return arrow at the end of the diagram? Same would go for the rest of the sequence diagrams.

![image](https://user-images.githubusercontent.com/53231968/97395678-693a2c00-1920-11eb-9c48-c181ece8587b.png)

dojh111 replied:
After the CreateSprintCommand command is completed, should the command object return control to the user? And maybe the second optional code block box can be smaller, since the other classes are not running any optional code in that period of time.

hughjazzman commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+- `toJson()`: Contains logic required to convert the model object into JSON string.  
+- `fromJson()`: Contains logic required to convert JSON object into its respective model class.    
+
+## Implementation
```
Perhaps there could be prefaces for these headers? Maybe even a diagram as well?

hughjazzman commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+When the Project is created, the duration of the Project and length of the Sprints are specified, thus, there will be a finite number of Sprints for each Project.
+
+
+Before execution:
```
Consider using formatting such as **bold** to differentiate the sections?

dojh111 commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+    * A new project is created, and added to project manager.
+
+#### Select Project
+### Task
```
The contents are well written, but maybe you could consider adding some sequence diagrams to provide a visual explanation on the sequence of events when the commands are run.

hughjazzman commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+#### Saving Data
+![Figure X: Saving Data](image/developerguide/storage_save.png "Saving Data")  
+
+Data will be saved under two scenarios: 1. When the program exits. 2. Changes made to the data. 
```
Consider putting the scenarios in new lines?

hughjazzman commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+- Conversion error due to missing properties.
+- Mapping error due to invalid property type (e.g. "name" is expecting a `String` but data read is an `Integer`).
+
+##### Converting and Mapping of JSON to Objects
```
Perhaps there could be a diagram for this section as well?

The header is also smaller than the text itself - could this be improved?

![image](https://user-images.githubusercontent.com/53231968/97396201-8e7b6a00-1921-11eb-884b-fbf04e87ad34.png)

dojh111 replied:
Perhaps a sequence diagram can be added to better illustrate the sequence of events happening here.

hughjazzman commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+    
 
-## Product scope
 ### Target user profile
```
Perhaps adding User Stories could be very useful to show the features being developed?

dojh111 commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+To make the data persistent and portable, JSON has been chosen as the format for data to be saved to a persistent storage such as storage drives, thumb drives and any other storage medium which stores the program. JSON is also **human-readable** which allows users to directly modify the data file easily. This can be useful in certain scenarios such as fixing the data file in the event of data cor ...
+
+#### Location
+![Figure X: Running the Jar](image/developerguide/savingdata1.png "Running the Jar")  
+_Figure X: Running the Jar_  
+![Figure X: Running in IDE](image/developerguide/savingdata2.png "Running in IDE")  
+_Figure X: Running in IDE_  
```
I really love these figures added. Gives a really good visual representation on where the saves are held.

hughjazzman commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,461 @@
+A task is created following the creation of a project, with a clear title, description 
+and priority of the task.
+Command executed by user `task /add -title <title> -desc <description> -priority <priority>`
+is passed, the following operations are implemented:
```
Consider adding a newline here so the bullet points render properly?

![image](https://user-images.githubusercontent.com/53231968/97396383-ef0aa700-1921-11eb-98a4-a3fe117c863c.png)

Same would go for all the sections.

hughjazzman reviewed (commented):
Nice project!

dojh111 reviewed (commented):
Generally, the content is concise and well written. Good Job! However, a big improvement is to add numbering to the sub-headings (3.1, 3.2.1 etc) which will improve the reading experience by a lot. The current diagrams are nice and detailed with minimal errors.

harryleecp commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,523 @@
+Each of the modules listed above are a collection of constituent classes, with each handling specialized tasks in-line with the SLAP principle.
+
+### UI Component
+![Figure X: Simplified class diagram for UI Component](./image/developerguide/UI.png "User Interface")  
```
Just a minor suggestion on details, maybe the word "UI" can be bolded so that there is a distinction between the diagram title and diagram components

harryleecp commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,523 @@
+    * `StorageHelpCommand`
+    * `TaskHelpCommand`
+### Model Component
+![Figure X: Simplified class diagram for Model Component](./image/developerguide/modelcomponent.png "Storage Component UML")  
```
The diagram itself is quite clear however a minor suggestion would be to standardize the location of component and operation titles within the diagram (i.e. Top left or top center)

harryleecp commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,523 @@
+When you start the program, the program will load the data file from its respective location and deserialise it into its respective objects. Data will be saved when the program exits or whenever the user makes changes to the program.  
+
+#### Loading Data
+![Figure X: Loading Data](image/developerguide/storage_load.png "Loading Data")  
```
One suggestion for this diagram would be to make sure that the all the arrowheads are touching the bar (same goes for Saving Data diagram)

harryleecp commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,523 @@
+## Implementation
+### Project
+#### Create Project
+![Figure X: Sequence diagram of CreateProjectCommand](./image/developerguide/createProjectSequenceDiagram.png
```
Some of the methods called included the dot sign such as "projectManager.addProject()", perhaps it should have been "addProject()" instead.

harryleecp reviewed (commented):
Great job with the developer guide as it is quite clear overall, however minor amendments are recommended for some of the mentioned diagrams.

chocomango committed:
Update on logger

chocomango committed:
Update on logger again

chocomango committed:
Update on logger again to test on github

chocomango committed:
Update on logger again to test on github again

chocomango committed:
Update on logger to print stack trace

chocomango committed:
Update on logger to print stack trace 2

chocomango committed:
Update on logger to print stack trace correctly

chocomango committed:
Update on logger to destroy after use

chocomango committed:
Final update on logger

homingjun committed:
Merge pull request #193 from chocomango/master

SprintCommand v2.1

homingjun committed:
Added JUnit tests for parser classes

homingjun committed:
Added JUnit tests for parser classes

homingjun committed:
Added JUnit tests for parser classes

homingjun committed:
Added JUnit tests for parser classes

homingjun committed:
Update ppp file name

homingjun committed:
Merge pull request #198 from homingjun/PPP1

Update ppp file name

homingjun committed:
Added logging for parser and help commands

homingjun committed:
Added logging for parser and help commands

keke101 committed:
Merge pull request #196 from keke101/storage-manager-v1.3

Fix storage load

e0426051 committed:
Many changes:
1. Apply SLAP to task classes
2. Fixes the null pointer exception
3. Add logging
4. Refactor code to ensure <30LOC per function
5. Add JavaDoc to many task functions
6. Refactor test code to minimize duplicate code

gmit22 committed:
Merge pull request #200 from homingjun/Logging

Added logging for parser and help commands

homingjun committed:
Fix merge conflict

e0426051 committed:
Merge branch 'master' into WeiLingLoggerTest

homingjun committed:
Fix merge conflict

homingjun committed:
Merge branch 'master' into ParserJUnit

homingjun committed:
Remove irrelevant log file

homingjun committed:
Remove irrelevant log file

homingjun committed:
Merge branch 'ParserJUnit' of https://github.com/homingjun/tp into ParserJUnit

e0426051 committed:
Git ignored.

homingjun committed:
Added log files to git ignore

e0426051 committed:
Merge branch 'WeiLingLoggerTest' of https://github.com/e0426051/tp into WeiLingLoggerTest

# Conflicts:
#	logs/scrum.log

keke101 committed:
Merge pull request #197 from homingjun/ParserJUnit

Added JUnit tests for parser classes

e0426051 committed:
Merge branch 'master' into WeiLingLoggerTest

e0426051 committed:
Merge pull request #201 from e0426051/WeiLingLoggerTest

Many changes, for Task V2.1

e0426051 committed:
Merge branch 'WeiLingLoggerTest'

keke101 committed:
Remove logfile

gmit22 committed:
logging added for Project, Member

gmit22 committed:
conflict resolved

gmit22 committed:
added logging for project and member

keke101 committed:
Update all LocalDate.now() usages
- All LocalDate.now() usages will reference to the clock given by Duke.getClock. (e.g. `LocalDate.now(Duke.getClock())`)
  - From now on, please reference to `Duke.getClock` whenever you use LocalDate or LocalDateTime's `now()`.
- This is to facilitate in automated text ui testing.
  - Allows us to set a fixed clock to ensure that dates do not always change

keke101 committed:
Add main class for text-ui testing
- `runtest.sh` and `runtest.bat` will run `TextUiTest.main()` in this class instead of `Duke.main()`

gmit22 committed:
Delete scrum.log

gmit22 committed:
Merge branch 'master' into projectLogging

keke101 committed:
Update script to use test main class
- Script will not be using the `Duke.main()`.
- It will be using `TextUiTest.main()` instead

keke101 committed:
Fix some message and javadoc

keke101 committed:
Update test cases

keke101 committed:
Remove redundant files

keke101 committed:
Fix checkstyle

gmit22 committed:
Duplicate project adding

gmit22 committed:
passing test

gmit22 committed:
Merge branch 'projectLogging' of https://github.com/gmit22/tp into projectLogging

gmit22 committed:
javadoc

gmit22 committed:
Merge pull request #205 from gmit22/projectLogging

Project logging

gmit22 committed:
updated merging

gmit22 committed:
assertions fixed

gmit22 committed:
changes

gmit22 committed:
updated java doc

keke101 committed:
Merge branch 'master' into text-ui-testing

keke101 committed:
Add back entry in .gitignore

keke101 committed:
Change logger setup to invoke first
- Moved logger setup before StorageManager as it needs to use the logger

keke101 committed:
Fix saving issues
- Fixed unable to save when `data/` directory is deleted when the program is running.

keke101 committed:
Add logging to storage components

gmit22 committed:
checkstyle fixed

gmit22 committed:
create project testing

gmit22 committed:
Select project testing added

gmit22 committed:
view project testing

gmit22 committed:
list project test

gmit22 committed:
checkstyle passed

gmit22 committed:
add member test

gmit22 committed:
member testing added

gmit22 committed:
checkstyle fixed

gmit22 committed:
updated minor

e0426051 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-F11-4/tp

keke101 committed:
Merge pull request #208 from keke101/text-ui-testing

Update Text Ui Testing

keke101 committed:
Merge branch 'master' into storage-manager-v1.4

chocomango committed:
Add more error printing for catchDate

chocomango committed:
Update expected output for textUI test

chocomango committed:
Add support to handle interruption (CTRL+C) from user

chocomango committed:
Update log generation in different platform

chocomango committed:
Update program exit message

homingjun committed:
Update class diagrams for logic and project implementation

homingjun committed:
Merge pull request #207 from gmit22/ped

Duplicate project adding prevented

homingjun committed:
Merge pull request #209 from keke101/storage-manager-v1.4

Storage manager v1.4

homingjun committed:
Added javadoc comments

gmit22 committed:
ppp draft

gmit22 committed:
Merge branch 'master' into JUnit

gmit22 committed:
Merge pull request #212 from gmit22/PPP

ppp draft

keke101 committed:
Add more assertions to storage components

keke101 committed:
Remove invalid assertions

homingjun committed:
Update anchors in table of contents

homingjun committed:
Set theme jekyll-theme-cayman

keke101 committed:
Remove more invalid assertions

keke101 committed:
Enable assertions for automated testing

keke101 committed:
Fix checkstyle

homingjun committed:
Fix header inconsistencies

homingjun committed:
Merge branch 'UGv0.1' of https://github.com/homingjun/tp into UGv0.1

keke101 committed:
Merge pull request #214 from homingjun/UGv0.1

Fix anchors in UG table of contents

gmit22 committed:
Merge branch 'master' into JUnit

gmit22 committed:
Update build.gradle

keke101 committed:
Merge pull request #210 from gmit22/JUnit

J unit

e0426051 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-F11-4/tp

keke101 committed:
Merge pull request #215 from keke101/storage-manager-v1.4

Add and Remove Assertions

e0426051 committed:
Update DG

e0426051 committed:
PPP add draft

keke101 committed:
Merge pull request #213 from homingjun/JavaDoc

Added javadoc comments

keke101 committed:
Add PPP

homingjun committed:
Merge pull request #211 from homingjun/DG2

Update class diagrams for logic and sequence diagram for project implementation

keke101 committed:
Fix assertion

keke101 committed:
Merge pull request #217 from keke101/PPP

Add PPP

e0426051 committed:
Removed curved corners from Architecture diagram, Model component, Parser and Command component, and UI component

homingjun committed:
Update PPP

homingjun committed:
Delete createProjectSequenceDiagram.png

homingjun committed:
Merge pull request #219 from homingjun/PPP3

Update PPP

homingjun committed:
Add sequence diagram for create project

e0426051 committed:
Fix merge conflict

keke101 committed:
Update storage loading section
- Change loading error behaviour from exiting with code 1 to proceeding in empty state.

homingjun committed:
Merge pull request #216 from e0426051/master

Update DG

keke101 committed:
Update developer guide
- Change the header to bolded normal text

homingjun committed:
Fix format issues

keke101 committed:
Update README.md

homingjun committed:
Merge pull request #220 from homingjun/DG3

Add sequence diagram for create project

chocomango committed:
Merge pull request #221 from chocomango/errorPrinting

Update error printing for catchDate

keke101 committed:
Merge branch 'master' into doc-v2.1

chocomango committed:
Merge pull request #224 from chocomango/stacktrace

Add support to handle interruption (CTRL+C) from user

homingjun committed:
Update DG table of content anchors

homingjun committed:
Update DG table of content anchors

homingjun committed:
Set theme jekyll-theme-slate

homingjun committed:
Merge pull request #227 from homingjun/DGTOB

Update developer guide table of contents anchors

chocomango committed:
Merge pull request #226 from chocomango/log

Update log generation in different platform

keke101 committed:
Update minor word in DG

keke101 committed:
Merge branch 'master' into doc-v2.1

gmit22 committed:
ppp updated

gmit22 committed:
added dg

keke101 committed:
Add caption to image

keke101 committed:
Update images
- Loading sequence
- Saving sequence

keke101 committed:
Add loading of data diagram

keke101 committed:
Update parsing diagram
- To a higher resolution.

keke101 committed:
Add serialising diagram
- Serialise objects into string in JSON format when saving data.

keke101 committed:
Fix developer guide

keke101 committed:
Update developer guide
- Fix typo
- Fix typo in diagram
- Other minor changes

keke101 committed:
Merge pull request #222 from keke101/doc-v2.1

Doc v2.1

keke101 committed:
Update error message

keke101 committed:
Update constants name
- Capitalised them

keke101 committed:
Fix checkstyle
- Revert capitalisation of instance constants

keke101 committed:
Update test case

keke101 committed:
Update .gitignore

keke101 committed:
Rename packages
Renamed all to com.scrumptious

keke101 committed:
Update build.gradle

keke101 committed:
Update automated testing scripts
- Updated MANIFEST.MF too

keke101 committed:
Rename Duke to Scrumptious

keke101 committed:
Rename DukeTest to ScrumptiousTest

keke101 committed:
Update DeveloperGuide.md

keke101 committed:
Rename DukeException to ScrumptiousException

keke101 committed:
Rename Duke occurrences

keke101 committed:
Remove unused imports

keke101 committed:
Fix checkstyle

e0426051 committed:
Commit to resolve conflict from pulling from team repo

e0426051 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-F11-4/tp

# Conflicts:
#	docs/image/developerguide/commandClassDiagram.png

e0426051 committed:
Fixed the UI part of DG

chocomango committed:
Add PPP for chocomango

keke101 committed:
Merge pull request #228 from keke101/rename-packages

Standardized packages name

e0426051 committed:
Merge pull request #229 from e0426051/master

Fix DG UI part

keke101 committed:
Update AboutUs.md

e0426051 committed:
Either PPP draft or PPP itself

e0426051 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-F11-4/tp

gmit22 committed:
Merge pull request #230 from chocomango/ppp

Add PPP for chocomango

e0426051 committed:
Merge pull request #231 from e0426051/master

PPP

gmit22 committed:
changes update

gmit22 committed:
Merge branch 'master' into PPP

gmit22 committed:
Merge pull request #232 from gmit22/PPP

Ppp

gmit22 committed:
changes

e0426051 committed:
PPP edit

e0426051 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-F11-4/tp

gmit22 committed:
Merge pull request #234 from gmit22/PPP

Deleted unnecessary testing

e0426051 committed:
edit

e0426051 committed:
Merge pull request #233 from e0426051/master

PPP edit

homingjun committed:
Update dg anchors to work on github pages and code

homingjun committed:
Fix minor issues

homingjun committed:
Merge pull request #235 from homingjun/DG4

Update DG anchors

e0426051 committed:
Update e0426051.md

keke101 committed:
Update README.md

keke101 committed:
Fix README.md

keke101 committed:
Merge pull request #236 from keke101/doc-v2.1

Update README

homingjun committed:
resize images to fit pdf

homingjun committed:
Set theme jekyll-theme-cayman

gmit22 committed:
Set theme jekyll-theme-cayman

homingjun committed:
Resize image

homingjun committed:
Merge branch 'images' of https://github.com/homingjun/tp into images

homingjun committed:
Merge pull request #237 from homingjun/images

Resize images to fit into pdf

gmit22 committed:
PPP final

gmit22 committed:
Merge pull request #238 from gmit22/PPP

PPP final

gmit22 committed:
updated ppp

homingjun committed:
Fix some header links

gmit22 committed:
formatting

gmit22 committed:
chang4e

homingjun committed:
Merge pull request #239 from homingjun/DG5

Fix some header links

gmit22 committed:
Merge pull request #240 from gmit22/PPP

Ppp

e0426051 committed:
Added basic assertions.

homingjun committed:
Fix spelling error

e0426051 committed:
Update e0426051.md

homingjun committed:
Merge pull request #242 from homingjun/DG5

Fix spelling error

e0426051 committed:
Attempt to fix assertions

gmit22 committed:
Update gmit22.md

e0426051 committed:
Merge pull request #241 from e0426051/master

Reinstated assertions.

gmit22 committed:
final done

gmit22 committed:
Merge pull request #243 from gmit22/PPP

final done

e0426051 committed:
Minor help edit

homingjun committed:
Added missing explanation to logic

e0426051 committed:
Update e0426051.md

homingjun committed:
Merge pull request #245 from homingjun/DG5

Added missing explanation to logic

e0426051 committed:
Merge pull request #244 from e0426051/master

Minor help edit

homingjun committed:
Update help command summary

e0426051 committed:
Update e0426051.md

homingjun committed:
Merge pull request #246 from homingjun/UG

Update help command summary

keke101 committed:
Update UserGuide

keke101 committed:
Update PPP

keke101 committed:
Fix UserGuide.md

keke101 committed:
Merge pull request #247 from keke101/doc-v2.1.1

Update Docs

keke101 committed:
Update DeveloperGuide.md

keke101 committed:
Update images to higher res

keke101 committed:
Merge pull request #248 from keke101/doc-v2.1.1

Update DeveloperGuide.md

keke101 committed:
Fix wrong image

e0426051 committed:
Update Sequence Diagram

keke101 committed:
Merge pull request #249 from keke101/doc-v2.1.1

Fix wrong image

e0426051 committed:
Merge pull request #250 from e0426051/master

Update Sequence Diagram

homingjun committed:
Finalise PPP

homingjun committed:
Finalise PPP

homingjun committed:
Finalise PPP

homingjun committed:
Finalise PPP

homingjun committed:
Merge pull request #251 from homingjun/PPPFinal

PPP Final

homingjun committed:
Add profile photo to aboutus

homingjun committed:
Add profile photo to aboutus

homingjun committed:
Add profile photo to aboutus

keke101 committed:
Fix PPP
- Reduce the page length

homingjun committed:
Merge pull request #252 from homingjun/PPPFinal

Add profile photo to AboutUs

keke101 committed:
Update PPP

gmit22 committed:
added image

gmit22 committed:
added img

keke101 committed:
Fix PPP
- Reduce length

gmit22 committed:
Merge pull request #254 from gmit22/PPP

About us image added

keke101 committed:
Merge branch 'master' into doc-v2.1.1

keke101 committed:
Update AboutUs
- Add profile picture

keke101 committed:
Merge pull request #253 from keke101/doc-v2.1.1

Update PPP and AboutUs.md

keke101 committed:
Fix PPP

keke101 committed:
Merge pull request #255 from keke101/doc-v2.1.1

Fix PPP

chocomango committed:
Add comments for sprint commands.

chocomango committed:
Add inheritance for Sprint command Junit

chocomango committed:
Update on code quality for junit test

chocomango committed:
Update PPP

chocomango committed:
Merge pull request #256 from chocomango/finalise

Finalise
